moduleframework/tools/helpmd_lint.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+#
+# Meta test family (MTF) is a tool to test components of a modular Fedora:
+# https://docs.pagure.org/modularity/
+# Copyright (C) 2017 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# he Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Authors: Jan Scotka <jscotka@redhat.com>
+#
+from __future__ import print_function
+import os
+
+from moduleframework import module_framework
+from moduleframework import helpfile_linter
+from moduleframework import dockerlinter
+
+
+class HelpMDLinter(module_framework.AvocadoTest):
+    """
+    :avocado: enable
+
+    """
+
+    dp = None
+
+    def setUp(self):
+        # it is not intended just for docker, but just docker packages are
+        # actually properly signed
+        self.dp = dockerlinter.DockerfileLinter()
+        self.helpmd = helpfile_linter.HelpMDLinter()
+        if self.dp.dockerfile is None:
+            self.skip()
+
+    def test_helpmd_exists(self):
+        self.assertTrue(self.helpmd)
+
+    def test_helpmd_image_name(self):
+        container_name = self.dp.get_docker_specific_env("NAME")
+        if container_name:
+            self.assertTrue(self.helpmd.get_image_maintainer_name(container_name.split('=')[1]))
+
+    def test_helpmd_maintainer_name(self):
+        maintainer_name = self.dp.get_specific_label("maintainer")
+        if maintainer_name:
+            self.assertTrue(self.helpmd.get_image_maintainer_name(maintainer_name))
+
+    def test_helpmd_name(self):
+        self.assertTrue(self.helpmd.get_tag("NAME"))
+
+    def test_helpmd_description(self):
+        self.assertTrue(self.helpmd.get_tag("DESCRIPTION"))
+
+    def test_helpmd_usage(self):
+        self.assertTrue(self.helpmd.get_tag("USAGE"))
+
+    def test_helpmd_environment_variables(self):
+        self.assertTrue(self.helpmd.get_tag("ENVIRONMENT VARIABLES"))
+
+    def test_helpmd_security_implications(self):
+        self.assertTrue(self.helpmd.get_tag("SECURITY IMPLICATIONS"))
```

Comment:
how is this a requirement?

Reply:
I will modify it, so that if container expose some ports, then the ports  should be mentioned in this label.